library/core/src/intrinsics/simd.rs

```diff
@@ -470,4 +486,36 @@ extern "platform-intrinsic" {
     ///
     /// `T` must be a vector of integers.
     pub fn simd_cttz<T>(x: T) -> T;
+
+    /// Round up each element to the next highest integer.
+    ///
+    /// `T` must be a vector of floats.
+    pub fn simd_ceil<T>(x: T) -> T;
+
+    /// Round down each element to the next lowest integer.
+    ///
+    /// `T` must be a vector of floats.
+    pub fn simd_floor<T>(x: T) -> T;
+
+    /// Round each element to the closest integer.
+    /// Ties are resolving by rounding away from 0.
+    ///
+    /// `T` must be a vector of floats.
+    pub fn simd_round<T>(x: T) -> T;
+
+    /// Return the integer part of each element.
```

Comment:
Hmm, I think I'd prefer if these said "integer-valued float" instead of "integer" in the relevant places, if you don't mind? Or something like that. "Non-fractional"? I'm not picky on exactly how it is worded, just to clarify that no this isn't meant to be a type-changing transformation. ( yes, the signature is T -> T, but we sometimes have to ask if things are implementing what they are meant to implement, so. )

Reply:
Fair.

Btw, we don't seem to have an intrinsic for simd-round-ties-to-even?